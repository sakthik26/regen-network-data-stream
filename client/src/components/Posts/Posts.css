.container {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 5px;
  padding: 20px;
  margin: 10px 0;
  cursor: pointer;
  background-color: #fff;
}

.title {
  font-size: 18px;
  font-weight: bold;
}

.date {
  font-size: 16px;
  color: #777;
  margin-bottom: 15px;
}
