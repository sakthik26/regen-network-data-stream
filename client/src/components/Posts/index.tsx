import { Post } from '../../types';
import './Posts.css';

const Posts = ({ title, createdAt, onClick }: Post) => {
  const formatDate = (dateString: string) => {
    const options: Intl.DateTimeFormatOptions = {
      year: 'numeric',
      month: 'long',
      day: 'numeric',
    };
    return new Date(dateString).toLocaleDateString(undefined, options);
  };

  return (
    <div className="container" onClick={onClick}>
      <div className="title">{title}</div>
      <div className="date">{formatDate(createdAt)}</div>
    </div>
  );
};

export default Posts;
