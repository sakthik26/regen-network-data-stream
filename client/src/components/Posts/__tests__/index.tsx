
import { render, } from '@testing-library/react';
import Posts from '../index';

describe('Posts component', () => {
  const post = {
    title: 'Test Title',
    id: '1',
    createdAt: '2024-02-11T12:00:00Z',
    onClick: jest.fn(),
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<Posts {...post} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByText } = render(<Posts {...post} />);
    expect(getByText(post.title)).toBeInTheDocument();
    expect(getByText('February 11, 2024')).toBeInTheDocument();
  });
});