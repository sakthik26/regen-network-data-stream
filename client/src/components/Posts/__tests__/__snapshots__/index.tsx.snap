// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="title"
    >
      Test Title
    </div>
    <div
      class="date"
    >
      February 11, 2024
    </div>
  </div>
</DocumentFragment>
`;
